### PR TITLE
Feature/18 delete event

### PR DESCRIPTION
**Pull request by al-ros**
image

create Edit event logic
create Delete event logic
add creating event via calendar cell

### CODE REVIEW

File: src/components/Button/Button.jsx [hunk 1]
```diff
@@ -2,11 +2,12 @@ import PropTypes from 'prop-types'
 import cn from 'classnames'
 import './Button.scss'
 
-const Button = ({ onClick, label, className, disabled, block, ...rest }) =>
+const Button = ({ onClick, label, className, disabled, block, invisible, ...rest }) =>
```
Comment (roman-vabishchevych) [inline src/components/Button/Button.jsx]: You can remove invisible.
It's better to use 
```
<>
  {  isButtonShouldBeShown && <Button>...</Button> }
</>
```

File: src/components/Button/Button.jsx [hunk 2]
```diff
@@ -2,11 +2,12 @@ import PropTypes from 'prop-types'
...
     <button disabled={ disabled } 
         className={ cn('button', className, {
             'button--disabled': disabled,
-            'button--block': block 
+            'button--block': block,
+            'button--invisible': invisible
```
Comment (roman-vabishchevych) [inline src/components/Button/Button.jsx]:
```suggestion

```

File: src/components/ModalEvent/ModalEvent.jsx [hunk 3]
```diff
@@ -39,12 +45,23 @@ const ModalEvent = ({ event: eventValue, onCancel, onSubmit }) => {
...
+      alert('The event already exists. Change the date or edit the current event from the calendar.')
+    } else {
+      onSubmit(event)
+    }
+  }
 
   const renderModalFooter = () => (<>
     <Button className="modal-event__control" block label="Cancel" onClick={ onCancel }/>
-    <Button className="modal-event__control" block label="Confirm" disabled={ !isValid } onClick={ isEventConflict }/>
+    { !isNewEvent && isEditMode && <Button className="modal-event__control" block label="Delete" disabled={ !isValid } onClick={ () => onSubmit({subject: '', userName: event.userName, day: event.day, time: event.time}) }/>}
```
Comment (roman-vabishchevych) [inline src/components/ModalEvent/ModalEvent.jsx]: please turn the code into multiple lines
```
{
  !isNewEvent && isEditMode && <Button className="modal-event__control"
    block
    label="Delete"
    ...
    onClick={ () => ... } />
  
}
```

**Review by roman-vabishchevych — CHANGES_REQUESTED** [review]
there are minor comments

File: src/components/Button/Button.jsx (hunk 1 above)
Comment (al-ros) [reply src/components/Button/Button.jsx]: Done

File: src/components/Button/Button.jsx (hunk 2 above)
Comment (al-ros) [reply src/components/Button/Button.jsx]: Done

File: src/components/ModalEvent/ModalEvent.jsx (hunk 3 above)
Comment (al-ros) [reply src/components/ModalEvent/ModalEvent.jsx]: Done

**Review by roman-vabishchevych — APPROVED** [review]

**Pull request merged**